src/middlewares/basic-middlewares.ts

```diff
@@ -0,0 +1,39 @@
+import { Logger } from "@fwl/logging";
+import { Tracer } from "@fwl/tracing";
+import {
+  errorMiddleware,
+  httpsRedirectMiddleware,
+  loggingMiddleware,
+  Middleware,
+  rateLimitingMiddleware,
+  recoveryMiddleware,
+  tracingMiddleware,
+} from "@fwl/web/dist/middlewares";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { authMiddleware } from "./auth-middleware";
+import { sentryMiddleware } from "./sentry-middleware";
+import rateLimitingConfig from "@src/configs/rate-limiting-config";
+
+const noAuthBasicMiddlewares = (
+  tracer: Tracer,
+  logger: Logger,
+): Middleware<NextApiRequest, NextApiResponse>[] => [
+  tracingMiddleware(tracer),
+  rateLimitingMiddleware(tracer, logger, rateLimitingConfig),
+  recoveryMiddleware(tracer),
+  sentryMiddleware(tracer),
+  errorMiddleware(tracer),
+  loggingMiddleware(tracer, logger),
+  httpsRedirectMiddleware(tracer),
+];
+
+const basicMiddlewares = (
+  tracer: Tracer,
+  logger: Logger,
+): Middleware<NextApiRequest, NextApiResponse>[] => [
+  ...noAuthBasicMiddlewares(tracer, logger),
+  authMiddleware(tracer),
+];
+
+export { basicMiddlewares, noAuthBasicMiddlewares };
```

Comment:
This basically removes all the duplicated middlewares every where in the code base to replace it with a `basicMiddlewares` declaration.